pkg/crossplane/service_mariadb_database.go

```diff
@@ -256,5 +268,15 @@ func createCredentials(endpoint *Endpoint, username, password, database string)
 		"jdbcUrl":      fmt.Sprintf("jdbc:mysql://%s:%d/%s?user=%s&password=%s", endpoint.Host, endpoint.Port, database, username, password),
 	}
 
+	if metricsPort != "" {
+		creds["metrics"] = []string{
+			fmt.Sprintf("http://%s:%s/metrics/haproxy-0", endpoint.Host, metricsPort),
+			fmt.Sprintf("http://%s:%s/metrics/haproxy-1", endpoint.Host, metricsPort),
```

Comment:
From what I can see, the Helm chart PR (https://github.com/appuio/charts/pull/312/files) doesn't implement the HAproxy metrics endpoints. Might be worth to check whether those are required, and either adjust the Helm chart PR or this PR to keep everything consistent.